src/Executables/WriteCceWorldtubeCoordsToFile/WriteCceWorldtubeCoordsToFile.cpp

```diff
@@ -0,0 +1,79 @@
+#include <boost/program_options/value_semantic.hpp>
+#include <cstddef>
+#include <exception>
+#include <string>
+
+#include "NumericalAlgorithms/SphericalHarmonics/AngularOrdering.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/IO/StrahlkorperCoordsToTextFile.hpp"
+#include "Parallel/Printf/Printf.hpp"
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
+
+/*
+ * This executable is used for writing the collocation points of a CCE
+ * Strahlkorper to a text file. This is useful for other NR codes so they can
+ * write data to certain points and we do the necessary transformations.
+ */
+int main(int argc, char** argv) {
+  boost::program_options::options_description desc(
+      "Program for writing the collocation points (coordinates) of a worldtube "
+      "sphere that SpECTRE CCE is able to read and interpret to a "
+      "file.\nDetails about the output file:\n"
+      " * There are no header or comment lines\n"
+      " * Each point is written to a new line of the output file\n"
+      " * The delimiter for the x, y, z components of each point is a space\n"
+      " * The points are written in scientific notation\n"
+      " * The sphere is centered on the origin (0, 0, 0)");
+  desc.add_options()("help,h", "show this help message")(
+      "radius,r", boost::program_options::value<double>()->required(),
+      "Radius of the worltube.")(
+      "lmax,L", boost::program_options::value<size_t>()->required(),
+      "The spherical harmonic L of the surface. The surface will have (L + 1) "
+      "* (2L + 1) total points")(
+      "output_file,o", boost::program_options::value<std::string>()->required(),
+      "Output filename for the points. No extension will be added.")(
+      "force,f", boost::program_options::bool_switch(),
+      "Overwrite the output file if it already exists");
+
+  boost::program_options::variables_map vars;
+
+  boost::program_options::store(
+      boost::program_options::command_line_parser(argc, argv)
+          .options(desc)
+          .run(),
+      vars);
+
+  if (vars.contains("help")) {
+    Parallel::printf("%s\n", desc);
+    return 0;
+  }
+
+  for (const auto& option : {"radius", "lmax", "output_file"}) {
+    if (not vars.contains(option)) {
+      Parallel::printf("Missing option: %s\n\n%s", option, desc);
+      return 1;
+    }
+  }
+
+  const double radius = vars["radius"].as<double>();
+  const size_t l_max = vars["lmax"].as<size_t>();
+  const std::array<double, 3> center{0.0, 0.0, 0.0};
+  const std::string output_file = vars["output_file"].as<std::string>();
+  const bool overwrite = vars["force"].as<bool>();
+
+  try {
+    ylm::write_strahlkorper_coords_to_text_file(
+        radius, l_max, center, output_file, ylm::AngularOrdering::Cce,
+        overwrite);
+  } catch (const std::exception& exception) {
+    Parallel::printf("%s\n", exception.what());
+    return 1;
```

Comment:
I didn't know the best way to handle this. If there's an error like if the file already exists, the wallclock time that's printed is wrong (this isn't a charm exec), it prinds Node and Core = 0 which doesn't really make sense in this context, and it prints the file where the error happened (not super useful to outside users).

I debated having the `write_strahlkorper_coords_to_text_file` function return a `std::optional<std::string>` that is the error message and then handle the error printing here. Thoughts are welcome.

Reply:
I wouldn't worry about it. The main message is that the file already exists and as long as people can figure that out easily I think it's fine.